include/blmc_robots/solo12.hpp

```diff
@@ -0,0 +1,397 @@
+   * prior to any getter to have up to date data.
+   */
+  const Eigen::Ref<Vector12d> get_joint_torques()
+  {
+    return joint_torques_;
+  }
+
+  /**
+   * @brief get_joint_torques
+   * @return the target joint torques
+   * WARNING !!!!
+   * The method <acquire_sensors>"()" has to be called
+   * prior to any getter to have up to date data.
+
+   */
+  const Eigen::Ref<Vector12d> get_joint_target_torques()
+  {
+    return joint_target_torques_;
+  }
+
+  /**
+   * @brief get_joint_encoder_index
+   * @return the position of the index of the encoders a the motor level
+   * WARNING !!!!
+   * The method <acquire_sensors>"()" has to be called
+   * prior to any getter to have up to date data.
+   */
+  const Eigen::Ref<Vector12d> get_joint_encoder_index()
+  {
+    return joint_encoder_index_;
+  }
+
+  /**
+   * @brief get_contact_sensors_states
+   * @return the state of the contacts states
+   * WARNING !!!!
+   * The method <acquire_sensors>"()" has to be called
+   * prior to any getter to have up to date data.
+   */
+  const Eigen::Ref<Eigen::Vector4d> get_contact_sensors_states()
+  {
+    return contact_sensors_states_;
+  }
+
+  /**
+   * @brief get_slider_positions
+   * @return the current sliders positions.
+   * WARNING !!!!
+   * The method <acquire_sensors>"()" has to be called
+   * prior to any getter to have up to date data.
+   */
+  const Eigen::Ref<Eigen::Vector4d> get_slider_positions()
+  {
+    return slider_positions_;
+  }
+
+  /**
+   * Hardware Status
+   */
+
+  /**
+   * @brief get_motor_enabled
+   * @return This gives the status (enabled/disabled) of each motors using the
+   * joint ordering convention.
+   */
+  const std::array<bool, 12>& get_motor_enabled()
+  {
+    return motor_enabled_;
+  }
+
+  /**
+   * @brief get_motor_ready
+   * @return This gives the status (enabled/disabled) of each motors using the
+   * joint ordering convention.
+   */
+  const std::array<bool, 12>& get_motor_ready()
+  {
+    return motor_ready_;
+  }
+
+  /**
+   * @brief get_motor_board_enabled
+   * @return This gives the status (enabled/disabled of the onboard control cards).
+   */
+  const std::array<bool, 6>& get_motor_board_enabled()
+  {
+    return motor_board_enabled_;
+  }
+
+  /**
+   * @brief get_motor_board_errors
+   * @return This gives the status (enabled/disabled of the onboard control cards).
+   */
+  const std::array<int, 6>& get_motor_board_errors()
+  {
+    return motor_board_errors_;
+  }
+
+  /**
+   * @brief has_error
+   * @return Returns true if the robot hardware has an error, false otherwise.
+   */
+  bool has_error() const
+  {
+    for (const auto& error_code : motor_board_errors_)
+    {
+      if (error_code != 0) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+private:
+  /**
+   * Joint properties
+   */
+
+  /** @brief Motors inertia. */
+  Vector12d motor_inertias_;
+  /** @brief DCM motor torque constants. */
```

Comment:
Want to have give the declarations here some more space and add a new line before the next docstring etc? Applies also to lines below.